packages/cli/src/commands/build.ts

```diff
@@ -608,6 +651,59 @@ async function doBuild(
   );
 }
 
+export async function determineFrameworkVersion(
+  frameworkRecord: UnboxPromise<ReturnType<typeof detectFrameworkRecord>>,
+  cwd: string,
+  output: Output
+) {
+  if (!frameworkRecord?.versionDependencies?.length) {
+    return;
+  }
+
+  for (let packageName of frameworkRecord.versionDependencies) {
+    const version = await lookupInstalledVersion(cwd, packageName, output);
+    if (version) {
+      return {
+        version,
+      };
+    }
+  }
+}
+
+interface NpmLSOutput {
+  dependencies?: {
+    [dependency: string]:
+      | {
+          version: string;
+        }
+      | undefined;
+  };
+}
+
+async function lookupInstalledVersion(
+  cwd: string,
+  packageName: string,
+  output: Output
+) {
+  try {
+    const args = `ls ${packageName} --depth=0 --json`.split(' ');
+    const { stdout } = await execa('npm', args, { cwd });
```

Comment:
How about reading package.json instead?

```
return require(`${packageName}/package.json`).version
```

Reply:
Huh, yeah. I never reach for dynamic require as a tool, but I should definitely consider it more often. This would work regardless of package manager, which would be great.

Reply:
This does seem to work. I [tested it in staging](https://github.com/vercel/api/pull/15601#issuecomment-1347192835).

I could not write a test for this, though. I would always get package not found. I tried manipulating `module.paths`, setting `NODE_PATH`, and using `process.chdir`, but nothing worked for me.